src/wrapped_radix_engine_toolkit/low_level/compiled_notarized_intent.rs

```diff
@@ -23,30 +23,36 @@ impl CompiledNotarizedIntent {
         }
     }
 
-    pub fn decompile(&self) -> Result<NotarizedTransaction> {
-        RET_decompile_notarize_tx(self.secret_magic.bytes())
-        .map_err(|e| {
-            error!("Failed to decompile bytes into Notarized Transaction, error: {:?}", e);
-            CommonError::FailedToDecompileBytesIntoNotarizedTransaction
-        })
-        .and_then(TryInto::<NotarizedTransaction>::try_into)
+    pub fn bytes(&self) -> BagOfBytes {
+        self.secret_magic.clone()
     }
-}
 
-impl TryFrom<ScryptoNotarizedTransaction> for NotarizedTransaction {
-    type Error = crate::CommonError;
+    pub fn decompile(&self) -> NotarizedTransaction {
```

Comment:
Right, so the thinking here is that the constructor of `CompiledNotarizedIntent` is hidden in Rust, Swift and Kotlin, so it is not possible to create an instance of it, thus, all instances are valid, since they have been created by calling `compile()` on `NotarizedTransaction` - where `NotarizedTransaction` itself is guaranteed to be valid.